test/core_tests.js

```diff
@@ -1,166 +1,266 @@
+    // Therefore we test them here, not in an event handler.
+    var notSupportedClass = iframeDoc.body.classList.contains( "impress-not-supported" );
+    var yesSupportedClass = iframeDoc.body.classList.contains( "impress-supported" );
+    if ( !_impressSupported() ) {
+      assert.ok( notSupportedClass,
+                 "body.impress-not-supported class still there." );
+      assert.ok( !yesSupportedClass,
+                 "body.impress-supported class was NOT added." );
+    } else {
+      assert.ok( !notSupportedClass,
+                 "body.impress-not-supported class was removed." );
+      assert.ok( yesSupportedClass,
+                 "body.impress-supported class was added." );
+
+      assert.ok( !iframeDoc.body.classList.contains( "impress-disabled" ),
+                 "body.impress-disabled is removed." );
+      assert.ok( iframeDoc.body.classList.contains( "impress-enabled" ),
+                 "body.impress-enabled is added." );
+
+      var canvas = iframeDoc.querySelector( "div#impress > div" );
+      assert.ok( !canvas.classList.contains( "step" ) && canvas.id === "",
+                 "Additional 'canvas' div inserted between div#impress root and steps." );
+      assert.equal( canvas.style.transform,
+                    "rotateZ(0deg) rotateY(0deg) rotateX(0deg) translate3d(1000px, 0px, 0px)",
+                    "canvas.style.transform initialized correctly" );
+      assert.equal( canvas.style.transformOrigin,
+                    "left top 0px",
+                    "canvas.style.transformOrigin initialized correctly" );
+      assert.equal( canvas.style.transformStyle,
+                    "preserve-3d",
+                    "canvas.style.transformStyle initialized correctly" );
+      assert.equal( canvas.style.transitionProperty,
+                    "all",
+                    "canvas.style.transitionProperty initialized correctly" );
+      assert.equal( canvas.style.transitionTimingFunction,
+                    "ease-in-out",
+                    "canvas.style.transitionTimingFunction initialized correctly" );
+
+      assert.equal( iframeDoc.documentElement.style.height,
+                    "100%",
+                    "documentElement.style.height is 100%" );
+
+      // Steps initialization
+      var step1 = iframeDoc.querySelector( "div#step-1" );
+      assert.equal( step1.style.position,
+                    "absolute",
+                    "Step position is 'absolute'." );
+
+      assert.ok( step1.classList.contains( "active" ),
+                 "Step 1 has active css class." );
+
+    }
+    doneSync();
+    console.log( "End init() test (sync)" );
+  } ); // LoadIframe()
+
+} );
+
+// Note: Here we focus on testing the core functionality of moving between
+// steps, the css classes set and unset, and events triggered.
+// TODO: more complex animations and check position, transitions, delays, etc...
+QUnit.test( "Impress Core API", function( assert ) {
```

Comment:
Generally I would prefer the tests to be smaller (focused on one particular functionality or method) rather than one big test for whole 'core API', but I understand the complexity of the tests setup etc - and I suppose nature of impress.js doesn't make it very unit-test-friendly.

So I'm OK to leave it as it is for now not to block further merges, just wanted to leave it as a note for future consideration.

Reply:
Current breakdown is:
* init() tests
* next(), prev(), goto() = "movement" tests
* navigation module (testing key events and clicking a link with syn)
* navigation module no-op tests (e.g. clicking on current slide should not cause movement)

They are grouped into 2 files. They could be grouped into 4, but I don't know that that's what you mean.

The setup overhead is indeed quite high. So I don't see it meaningful to think that each "QUnit.test()" section would only test a single thing, I've rather used that as a kind of module, which will then do 30-100 different "actual tests" and related assertions.